feat: UserSkel improvements (`firstname`, `lastname`, `sync`) and Google Auth user information synchronization

phorward opened the pull request:
This should become standard to allow for better pre-defined personalization.

sveneberth commented:
Okay for me, but fill a new skel with the information from the google login
https://github.com/viur-framework/viur-core/blob/209396bf72a1fea6fdffac13c825397b50728b71/core/modules/user.py#L448

sveneberth reviewed (changes requested):
Please fill them with values, as in https://github.com/viur-framework/viur-core/pull/677

phorward commented:
> Please fill them with values, as in #677

you mean #2?!

phorward commented:
@sveneberth I've added a `synced`-Flag to let the user decide if information should be kept synchronous with Google.
I've also changed the way how the `update`-flag is being determined.

sveneberth commented on core/modules/user.py:
```diff
@@ -47,6 +57,16 @@ class UserSkel(skeleton.Skeleton):
         unique=UniqueValue(UniqueLockMethod.SameValue, False, "UID already in use"),
     )
 
+    synced = BooleanBone(
```
```suggestion
    sync = BooleanBone(
```
Please rename this bone to `sync`.

sveneberth commented on core/modules/user.py:
```diff
@@ -469,24 +489,43 @@ def login(self, skey="", token="", *args, **kwargs):
+            update = True
+
+        # Take user information from Google, if wanted!
+        if userSkel["synced"]:
+            logging.debug(f"""{userSkel["synced"]=}""")
```
```suggestion
        if userSkel["sync"]:
```

sveneberth commented on core/modules/user.py:
```diff
@@ -469,24 +489,43 @@ def login(self, skey="", token="", *args, **kwargs):
+                "firstname": userInfo.get("given_name"),
+                "lastname": userInfo.get("family_name"),
+            }.items():
+                logging.debug(f"""{target=} {source=}""")
```
```suggestion
```

sveneberth commented on core/modules/user.py:
```diff
@@ -469,24 +489,43 @@ def login(self, skey="", token="", *args, **kwargs):
+                    userSkel[target] = source
+                    update = True
+
+        logging.debug(f"{update=}")
```
```suggestion
```
